File: Solutions/1313. Decompress Run-Length Encoded List/1313. Decompress Run-Length Encoded List.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int size=0;
        for(int i=0;i<nums.length;i+=2)
            size+=nums[i];
        int[] res=new int[size];
        size=0;
        for(int i=0,j=1;i<nums.length;i+=2,j+=2)
            while(nums[i]-->0)
                res[size++]=nums[j];
        return res;
    }
}

File: Solutions/1313. Decompress Run-Length Encoded List/1313. Decompress Run-Length Encoded List.cpp
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int> res;
        for(int i=0,j=1;i<nums.size();i+=2,j+=2)
            while(nums[i]--)
                res.push_back(nums[j]);
        return res;
    }
};